aerogear/managed-services-broker
# Provision fuse pods in a separate namespace from the user's namespace

## Description
Provision the fuse operator and deployment configs in a separate namespace from the user's namespace.

When the user provisions fuse from the service catalog, it will create a service instance in the user's namespace. The broker will then create a separate namespace, only accessible to an admin user, to deploy the operator and create the custom resource into.

## Progress
- [x] Create separate namespace for the fuse operator
- [x] Deploy Fuse operator into that namespace
- [x] Create Fuse custom resource into that namespace
- [x] Pass Fuse's dashboard url to the service instance created in the user's namespace
- [x] Allow the user who provisioned the service access to the Fuse dashboard

## Verification Steps

1. Create the Fuse [custom resource definition](https://github.com/syndesisio/syndesis/blob/master/install/operator/deploy/syndesis-crd.yml).
2. Follow the steps on how to [deploy the broker](https://github.com/aerogear/managed-services-broker/blob/provision-fuse/README.md#deploying-the-broker).
3. Verify that you can see the fuse plan in the service catalog.
4. Provision fuse.
5. Verify that a new namespace, only accessed by an admin user, has been created where the fuse operator and its deployment configs are created.
6. Verify that when provision completes, the dashboard url shows up on the service instance in the user namespace.
7. Verify that the user can access the dashboard url.
